controllers/k8ssandra/k8ssandracluster_controller.go

```diff
@@ -451,69 +189,71 @@ func (r *K8ssandraClusterReconciler) reconcileStargate(
 	logger = logger.WithValues("Stargate", stargateKey)
 
 	if stargateTemplate != nil {
+		logger.Info("Reconcile Stargate")
 
 		desiredStargate := r.newStargate(stargateKey, kc, stargateTemplate, actualDc)
-		desiredStargateHash := utils.DeepHashString(desiredStargate)
-		desiredStargate.Annotations[api.ResourceHashAnnotation] = desiredStargateHash
+		utils.AddHashAnnotation(desiredStargate, api.ResourceHashAnnotation)
 
 		if err := remoteClient.Get(ctx, stargateKey, actualStargate); err != nil {
 			if errors.IsNotFound(err) {
 				logger.Info("Creating Stargate resource")
 				if err := remoteClient.Create(ctx, desiredStargate); err != nil {
 					logger.Error(err, "Failed to create Stargate resource")
-					return ctrl.Result{}, err
+					return result.Error(err)
 				} else {
-					return ctrl.Result{RequeueAfter: r.ReconcilerConfig.DefaultDelay}, nil
+					return result.RequeueSoon(int(r.DefaultDelay))
```

Comment:
The cast to int everywhere is a bit disappointing, it would be nicer if `RequeueSoon` took a `time.Duration` directly. But this can be improved in a follow-up PR.